src/Compilers/Core/Portable/CodeGen/ItemTokenMap.cs

```diff
@@ -1,27 +1,24 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
-using System.Threading;
-using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CodeGen
 {
     /// <summary>
-    /// Handles storage of strings referenced via tokens in metadata. When items are stored 
-    /// they are uniquely "associated" with fake token, which is basically a sequential number.
-    /// IL gen will use these fake tokens during codegen and later, when actual token values 
-    /// are known the method bodies will be patched.
-    /// To support these two scenarios we need two maps - Item-->uint, and uint-->Item.  (the second is really just a list).
+    /// Handles storage of items referenced via tokens in metadata. When items are stored 
+    /// they are uniquely "associated" with fake tokens, which are basically sequential numbers.
+    /// IL gen will use these fake tokens during codegen and later, when actual values 
+    /// are known, the method bodies will be patched.
+    /// To support these two scenarios we need two maps - Item-->uint, and uint-->Item. (The second is really just a list).
     /// </summary>
-    internal sealed class StringTokenMap
+    internal sealed class ItemTokenMap<T> where T: class
```

Comment:
Probably, but I don't think making that change would have any significant effect. This type is instantiated only from PEModule builder. Switching to a struct might save a couple of dozen bytes per compilation.